### ISSUE TITLE
`resource_id` for `primary_knowledge_source` missing for TMKP edge on CI

### ISSUE THREAD

**Issue by andrewsu** [biothings/biothings_explorer#759]
When I run POST the query below to https://bte.ci.transltr.io/v1/query, I get an edge without a valid `resource_id` for `primary_knowledge_source`:

```
            "edges": {
                "8ebb881246f086e1e2ae593bad8f6872": {
                    "predicate": "biolink:affects",
                    "subject": "PUBCHEM.COMPOUND:5918",
                    "object": "NCBIGene:158",
                    "qualifiers": [
                    ],
                    "attributes": [
                    ],
                    "sources": [
                        {
                            "resource_role": "primary_knowledge_source"
                        },
                        {
                            "resource_id": "infores:text-mining-provider-targeted",
                            "resource_role": "aggregator_knowledge_source",
                            "upstream_resource_ids": [
                                null
                            ]
                        },
                        {
                            "resource_id": "infores:biothings-explorer",
                            "resource_role": "aggregator_knowledge_source",
                            "upstream_resource_ids": [
                                "infores:text-mining-provider-targeted"
                            ]
                        }
                    ]
                }
            }
        },
```

when I POST the same query to our prod instance https://bte.transltr.io/v1/query, I get a valid response.

```
            "edges": {
                "8ebb881246f086e1e2ae593bad8f6872": {
                    "predicate": "biolink:affects",
                    "subject": "PUBCHEM.COMPOUND:5918",
                    "object": "NCBIGene:158",
                    "qualifiers": [
                    ],
                    "attributes": [
                    ],
                    "sources": [
                        {
                            "resource_id": "infores:text-mining-provider-targeted",
                            "resource_role": "primary_knowledge_source"
                        },
                        {
                            "resource_id": "infores:biothings-explorer",
                            "resource_role": "aggregator_knowledge_source",
                            "upstream_resource_ids": [
                                "infores:text-mining-provider-targeted"
                            ]
                        }
                    ]
                }
            }
        },
```

initially assigning to @tokebe to assess whether this seems like something that might have broken in the recent refactoring.

Initially reported in https://github.com/NCATSTranslator/Feedback/issues/627 by @edeutsch 

<details>
<summary> TRAPI query </summary>

```
{
    "message": {
        "query_graph": {
            "edges": {
                "t_edge": {
                    "object": "on",
                    "predicates": [
                        "biolink:affects"
                    ],
                    "qualifier_constraints": [
                        {
                            "qualifier_set": [
                                {
                                    "qualifier_type_id": "biolink:object_aspect_qualifier",
                                    "qualifier_value": "activity_or_abundance"
                                },
                                {
                                    "qualifier_type_id": "biolink:object_direction_qualifier",
                                    "qualifier_value": "increased"
                                }
                            ]
                        }
                    ],
                    "subject": "sn"
                }
            },
            "nodes": {
                "on": {
                    "categories": [
                        "biolink:Gene"
                    ],
                    "ids": [
                        "NCBIGene:158"
                    ]
                },
                "sn": {
                    "categories": [
                        "biolink:ChemicalEntity"
                    ]
                }
            }
        }
    }
}
```
</details>

**Comment by colleenXu** [comment]
I wonder if this "reverted" behavior is related to https://github.com/biothings/biothings_explorer/issues/761 or https://github.com/biothings/biothings_explorer/issues/757...

**Comment by colleenXu** [comment]
Now that we've deployed code changes to CI, I get the desired response (edge-sources info is correctly formatted). I ran the example query in BTE CI (I used the `https://bte.ci.transltr.io/v1/smartapi/978fe380a147a8641caf72320862697b/query` to test Text-Mining Targeted directly).

But I think this was "fixed" by the smartapi-specs update in https://github.com/biothings/bte-server/pull/6. I'm not sure if the larger cache-update issues were fixed yet, given the [recent lab Slack convo](https://suwulab.slack.com/archives/CC218TEKC/p1702673534285579)
